firstapp/app/src/androidTest/java/com/example/firstapp/CUJExecutor.java

```diff
@@ -0,0 +1,40 @@
+package com.example.firstapp;
+import android.os.SystemClock;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject;
+import org.junit.Test;
+
+public class CUJExecutor {
+
+  
+
+    /**
+     * Executes a CUJ with preparatory actions pre and measured actions post. Caches information for later use.
+     */
+
+    public static void executeCUJ(String[] preCUJ, String[] postCUJ) throws Exception {
+        String[] CUJ = new String[preCUJ.length + postCUJ.length];
+        if (preCUJ != null) {
+            System.arraycopy(preCUJ, 0, CUJ, 0, preCUJ.length);
+        }
+        System.arraycopy(postCUJ, 0, CUJ, preCUJ.length, postCUJ.length);
+        String pkg = CUJ[0];
+        ShellUtility.forceQuitApp(pkg);
+
+        String[][] cached_tokens = new String[CUJ.length][];
+        UiObject[] cached_objects = new UiObject[CUJ.length];
```

Comment:
1. camel-case:

    cached_tokens -> cachedTokens
    cached_objects -> cachedObjects

2. I don't see them being used anywhere.  How will they be used?

Reply:
will be used later when we run through the CUJ more than once